### ISSUE TITLE
Usage of ground truth poses and pose priors (and some minor stuff)

### ISSUE THREAD

**Issue by humenbergerm** [openMVG/openMVG#1264]
Hi Pierre!

I work with OpenMVG for a while now (thanks for the great work) and I have implemented some additional features and some small modifications I would like to share with you and the community. The main goal of my work was to compute an SfM model using ground truth (assumed to be accurate; similar to #1195) or prior (assumed to be noisy) poses. Please have a look at the things I did so far (see list below) and let me know if any of this is interesting for you. If yes, I am happy to generate a pull request.

Furthermore, I am interested in improving the use of rotation priors since right now I just skip the global rotation estimation step. I was hoping you have some suggestions and ideas about that.

- added new sensor (Sony RX0) to camera database
- save and load sfm_data:
  - the flags "use_pose_center_prior" and "use_pose_rotation_prior" are always stored now
  - prior data is only loaded if these flags are set "true"
- enable rotation prior for global sfm pipeline by skipping global rotations estimation: activated by "use_pose_rotation_prior" flags
- data can be appended to an already existing sfm_data file (e.g. if several cameras are used and not all images are in the same folder)
- save and load matches:
  - if 'local_path' is set in sfm_data file, the .feat and .desc files are stored in (and loaded from) a subfolder named after the directory given in 'local_path'
    - e.g.: matches_dir=/home/user/matches, local_path=camera1 -> features and descriptors are stored in /home/user/matches/camera1
    - this is useful (but not mandatory) if several multiple image folders are used within one sfm_data file
- added option 'matchesFileNamePrefix' to 'main_ComputeMatches': all matches files will have this prefix (e.g.: matchesFileNamePrefix=test -> test_matches.e.bin)

Best,
Martin

**Comment by Yannnnnnnnnnnn** [comment]
@humenbergerm I think what we do are almost the same  about the gt ^-^.

**Comment by pmoulon** [comment]
@humenbergerm, @Yannnnnnnnnnnn  I'm working on the PR #1195
I'm checking that every importer is working as expected and fixing things on the go.
See some WIP https://github.com/openMVG/openMVG/pull/1195#issuecomment-378487557
I'm almost done, so the PR will be merged soon.

@humenbergerm I did not got the time yet to look in detail to your message but I will do soon.

**Comment by Yannnnnnnnnnnn** [comment]
@humenbergerm Yep, I almost done the ground truth loading part and support five dataset now. 
We can work together to support more dataset.

However after I loaded the ground truth, I will not do the BA again. I just simply use the main_ComputeStructureFromKnownPoses to rebuild the scene. 
So i din't look into the SfM with rotation priors.

**Comment by humenbergerm** [comment]
@Yannnnnnnnnnnn @pmoulon Cool, I will then wait till you guys finished #1195.

Best,
Martin
